be/src/exec/pipeline/analysis/analytic_sink_operator.cpp

```diff
@@ -84,6 +84,12 @@ void AnalyticSinkOperator::close(RuntimeState* state) {
 
 Status AnalyticSinkOperator::set_finishing(RuntimeState* state) {
     _is_finished = true;
+
+    // skip processing if cancelled
+    if (state->is_cancelled()) {
+        return Status::Cancelled("runtime state is cancelled");
+    }
+
     _analytor->input_eos() = true;
```

Comment:
this state should be set before cancelled return?

Reply:
not necessary. cancel has its indenpendent processing.